exercise/exercise_models.py

```diff
@@ -704,7 +704,9 @@ def grade(self, request, submission, no_penalties=False, url_name="exercise"):
         Loads the exercise feedback page.
         """
         # Get the language from the submission
-        language = submission.lang or get_language()
+        language = submission.lang
+        if language is None:
+            language = submission.meta_data['lang'] = get_language()
```

Comment:
It would make more sense to set this in the location, which stores `meta_data` for the first time. Even if this function is executed as part of the original request, it's likely that it will change.

Reply:
I thought it would be safer to do this way, do you mean it would be enough to only do this in `create_from_post`?
https://github.com/apluslms/a-plus/blob/73aea9db45f186da052957024500d44ea6932274/exercise/submission_models.py#L38-L45

Reply:
Yeah. I think it should be part of that function. Though, that function shouldn't be part of the query manager object, but we'll go with that for now.